packages/flutter/lib/src/rendering/sliver_group.dart

```diff
@@ -213,7 +214,15 @@ class RenderSliverMainAxisGroup extends RenderSliver with ContainerRenderObjectM
         double childScrollOffset = 0.0;
         RenderSliver? current = childBefore(child as RenderSliver);
         while (current != null) {
-          childScrollOffset += current.geometry!.scrollExtent;
+          // If the current sliver is not the first child, add twice the scrollExtent to the scrollOffset
+          // to account for both the current and the previous slivers.
+          // If the current sliver is the first child, add its scrollExtent once to the scrollOffset
+          // to account for only the current sliver.
+          if (childBefore(current) != null) {
+            childScrollOffset += 2 * current.geometry!.scrollExtent;
```

Comment:
2 feels like a magic number here. Shouldn't it depends on the preceding child's size?

Reply:
Yes, you're right. I adjusted the code to have the expected behavior and avoid 2 as the magic number, in addition to preventing it from scrolling further than expected.